nx-X11/programs/Xserver/hw/nxagent/NXrender.c

```diff
@@ -1001,17 +1001,20 @@ ProcRenderCompositeGlyphs (ClientPtr client)
 	listsBase = (GlyphListPtr) malloc (nlist * sizeof (GlyphListRec));
 	if (!listsBase)
 	{
-	    free(glyphsBase);
+	    if (glyphsBase != glyphsLocal)
+		free(glyphsBase);
 	    return BadAlloc;
```

Comment:
This matches X.org commit bc61787a20e7683cbc4dfa45fe855da98a8c0cd0 (with some little diversion). I'd say, let's port that commit here (and have that documented in git log).

Reply:
> This matches X.org commit bc61787a20e7683cbc4dfa45fe855da98a8c0cd0 (with some little diversion). I'd say, let's port that commit here (and have that documented in git log).

Ping on this...

Reply:
ok, will have a look on this.

Reply:
> ok, will have a look at this.

any news on this?

Reply:
Well, there's a bit more to so. The return code stuff has been changed meanwhile (92ed75ac59e2d3af149cddb962efd05fc8487750) so the code would differ anyway. For now I would like to keep it that way, it will be redone once I go on with updating the codebase to a newer Xorg level.